src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java

```diff
@@ -545,69 +545,47 @@ private void populateSummaryStatisticsFromResponses(
             String recipientEmail = response.recipient;
 
             // Compute number of responses including user's self response
-            if (!numResponses.containsKey(recipientEmail)) {
-                numResponses.put(recipientEmail, 0);
-            }
-            int numOfResponses = numResponses.get(recipientEmail) + 1;
+            int numOfResponses = numResponses.getOrDefault(recipientEmail, 0) + 1;
             numResponses.put(recipientEmail, numOfResponses);
 
             // Compute number of responses excluding user's self response
-            if (!numResponsesExcludingSelf.containsKey(recipientEmail)) {
-                numResponsesExcludingSelf.put(recipientEmail, 0);
-            }
             boolean isSelfResponse = giverEmail.equalsIgnoreCase(recipientEmail);
             if (!isSelfResponse) {
-                int numOfResponsesExcludingSelf = numResponsesExcludingSelf.get(recipientEmail) + 1;
+                int numOfResponsesExcludingSelf = numResponsesExcludingSelf.getOrDefault(recipientEmail, 0) + 1;
```

Comment:
Be careful, this is not the same. What happens if isSelfResponse is true?
(unless you have thoroughly checked the data flow for all cases; in which case, do elaborate)

Reply:
My bad. So if `isSelfResponse` is `false` the key value pair would not be put into the map.